### DISCUSSION TITLE
micropython-lib aiorepl not working on STM32 Port

### DISCUSSION THREAD

**Discussion by rootfrogsMakes** [micropython/micropython#11788] [STM32 / Pyboard]
Hello,
I am trying to get the [aiorepl](https://github.com/micropython/micropython-lib/tree/master/micropython/aiorepl) to run on the [STM32F4DISC](https://micropython.org/download/STM32F4DISC/) board.
If I try this, I get the following output:

```
Starting tasks...
Starting asyncio REPL...
Traceback (most recent call last):
  File "main.py", line 27, in <module>
  File "uasyncio/core.py", line 1, in run
  File "uasyncio/core.py", line 1, in run_until_complete
  File "uasyncio/core.py", line 1, in run_until_complete
  File "main.py", line 25, in main
  File "uasyncio/funcs.py", line 1, in gather
  File "uasyncio/core.py", line 1, in run_until_complete
  File "aiorepl.py", line 91, in task
AttributeError: 'module' object has no attribute '__dict__'
MicroPython v1.19.1 on 2022-06-18; F4DISC with STM32F407
Type "help()" for more information.
```

line 91 of aiorepl is the following:
```
g = __import__("__main__").__dict__
```

In the REPL I see that there is no __dict__ in main:
```
>>> dir(__import__("__main__"))
['__class__', '__name__', 'machine', 'pyb']
```

So the aiorepl lib really cannot run with this port.
What can I do to make aiorepl work on this Port?

**Comment by jimmo** [comment]
I wrote `aioble` using an STM32 board (pyboard) as the target.

> MicroPython v1.19.1 on 2022-06-18; F4DISC with STM32F407

Support for `__dict__` on modules was added in v1.20.

**Comment by rootfrogsMakes** [comment]
Oh, thanks.
I had not noticed that there is a new version of micropython. So I used v1.20 aiorepl with v1.19 of MicroPython.
I updated to MicroPython v1.20 and now it works.
Thanks a lot.
